scene/2d/joint_2d.cpp

```diff
@@ -62,7 +62,9 @@ void Joint2D::_update_joint(bool p_only_free) {
 	bb = RID();
 	configured = false;
 
-	if (p_only_free || !is_inside_tree()) {
+	bool disabled = is_inside_tree() && !is_enabled() && disable_mode == DISABLE_MODE_REMOVE;
```

Comment:
```suggestion
	bool disabled = !is_enabled() && disable_mode == DISABLE_MODE_REMOVE;
```
Already covered by the or block, this is only considered if `!is_inside_tree() == false`